web/src/reducers/activities.js

```diff
@@ -46,6 +51,29 @@ const reducer = (state = initialState, action) => {
         },
         state
       );
+    case ADD_ACTIVITY_MILESTONE:
+      return u(
+        {
+          byId: {
+            [action.id]: {
+              milestones: milestones => [...milestones, newMilestone()]
+            }
+          }
+        },
+        state
+      );
+    case REMOVE_ACTIVITY_MILESTONE:
+      return u(
+        {
+          byId: {
+            [action.id]: {
+              milestones: milestones =>
+                milestones.filter((_, i) => i !== action.milestoneIdx)
```

Comment:
Interesting.  I'd always done this with `splice`.  I'm guessing there's a performance difference between the two, but whatever it might be, it's irrelevant at this scale of data.  And I think this `filter` approach is more readable.  👍